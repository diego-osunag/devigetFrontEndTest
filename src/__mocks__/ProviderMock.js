import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import stateMock from './stateMock';
import reducer from '../reducers';

const store = createStore(reducer, stateMock);

const ProviderMock = (props) => (
  <Provider store={store}>
    {props.children}
  </Provider>
);

export default ProviderMock;
