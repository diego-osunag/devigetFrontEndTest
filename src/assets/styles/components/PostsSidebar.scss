.postsSidebar {
  background-color: #1E1E1E;
  width: 435px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100vh;
  transition: margin 350ms;
  flex-shrink: 0;
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}

.postsSidebar__title {
  color: white;
  background-color: #3C3C3C;
  font-weight: normal;
  text-align: center;
  padding: 50px 0 20px 0;
  margin: 0;
}

.postsSidebar__dismissAll {
  color: #f3bb83;
  background-color: #222;
  font-weight: normal;
  text-align: center;
  padding: 18px 0;
  margin: 0;
  cursor: pointer;
  transition: background-color 400ms;
}

.postsSidebar__dismissAll:hover {
  background-color: #333;
}

.postsSidebar__dismissAll:active {
  background-color: #444;
}

.postsDetail__icon {
  width: 20px;
  position: fixed;
  right: 20px;
  top: 20px;
  display: none;
  cursor: pointer;
  transition: transform 200ms;
}

.postsDetail__icon:active {
  transform: scale(1.3);
}

.postsDetail__icon--reload {
  width: 25px;
  position: fixed;
  right: 20px;
  bottom: 20px;
  cursor: pointer;
  transition: transform 200ms;
}

.postsDetail__icon--reload:active {
  transform: scale(1.3);
}
