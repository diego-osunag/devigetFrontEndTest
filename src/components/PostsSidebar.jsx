/* eslint-disable jsx-quotes */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { dismissAllPosts, loadPosts, removeSelectedPost, updatePageCount } from '../actions';
import PostsList from './PostsList';
import '../assets/styles/components/PostsSidebar.scss';
import menuIcon from '../assets/static/hamburguer-menu-icon.png';
import reloadIcon from '../assets/static/reload.png';
import '../swiped-events.min';

class PostsSidebar extends Component {

  fetchApiData() {
    const API = 'https://www.reddit.com/r/reactjs/top.json';
    fetch(API)
      .then((response) => response.json())
      .then((responseData) => {
        const posts = responseData.data.children.map(item => (item.data));
        this.props.loadPosts(posts);
        this.props.updatePageCount(Math.ceil(this.props.postsList.length / this.props.perPage));
      });
  }

  handleClick = () => {
    this.props.dismissAllPosts();
    this.props.removeSelectedPost();
    this.props.updatePageCount(0);
  }

  handleRefresh = () => {
    this.props.dismissAllPosts();
    this.props.removeSelectedPost();
    this.fetchApiData();    
  }

  toggleSideBar = () => {
    const sidebar = document.getElementById('sideBar');
    sidebar.classList.toggle('visible');
  }

  showSideBar = () => {
    const sidebar = document.getElementById('sideBar');
    if (!sidebar.classList.contains('visible')) sidebar.classList.toggle('visible');
  }

  hideSideBar = () => {
    const sidebar = document.getElementById('sideBar');
    if (sidebar.classList.contains('visible')) sidebar.classList.toggle('visible');
  }

  handleToggle = () => {
    const sidebar = document.getElementById('sideBar');
    sidebar.classList.toggle('visible');
  }

  handleSwipe() {
    document.addEventListener('swiped-right', (event) => {
      this.showSideBar();
    });
    document.addEventListener('swiped-left', (event) => {
      this.hideSideBar();
    });
  }  

  componentDidMount() {
    this.fetchApiData();
    this.handleSwipe();
  }

  render() {
    return (
      <section id="sideBar" className="postsSidebar">
        <h2 className="postsSidebar__title">Top Reddit Posts (/r/reactjs)</h2>
        <PostsList />
        <h3 className="postsSidebar__dismissAll" onClick={this.handleClick}>
          Dismiss All
        </h3>
        <img id='btn' className="postsDetail__icon" src={menuIcon} onClick={this.handleToggle} alt='' />
        <img className="postsDetail__icon--reload" src={reloadIcon} onClick={this.handleRefresh} alt='' />
      </section>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    postsList: state.postsList,
    pageCount: state.pageCount,
    perPage: state.perPage,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    dismissAllPosts: () => dispatch(dismissAllPosts()),
    loadPosts: (posts) => dispatch(loadPosts(posts)),
    removeSelectedPost: () => dispatch(removeSelectedPost()),
    updatePageCount: (number) => dispatch(updatePageCount(number)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(PostsSidebar);
