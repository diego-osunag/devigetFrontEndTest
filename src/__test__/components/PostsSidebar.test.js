import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import PostsSidebar from '../../components/PostsSidebar';

describe('<PostsSidebar />', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });
  const post = mount(
    <ProviderMock>
      <PostsSidebar />
    </ProviderMock>,
  );

  test('PostsSidebar component rendering', () => {
    expect(post.length).toEqual(1);
  });

  test('Title rendering', () => {
    expect(post.find('h2.postsSidebar__title').text()).toEqual('Top Reddit Posts (/r/reactjs)');
  });

  test('Dismiss All button rendering', () => {
    expect(post.find('h3.postsSidebar__dismissAll').text()).toEqual('Dismiss All');
  });

  test('Menu icon rendering', () => {
    expect(post.find('img.postsDetail__icon').length).toEqual(1);
  });

  test('Reload icon rendering', () => {
    expect(post.find('img.postsDetail__icon--reload').length).toEqual(1);
  });

});

describe('<PostsSidebar /> Snapshot', () => {
  test('PostsSidebar UI rendering correctly', () => {
    const post = create(
      <ProviderMock>
        <PostsSidebar />
      </ProviderMock>,
    );
    expect(post.toJSON()).toMatchSnapshot();
  });
});