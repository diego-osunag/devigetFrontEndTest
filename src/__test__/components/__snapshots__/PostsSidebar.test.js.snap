// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsSidebar /> Snapshot PostsSidebar UI rendering correctly 1`] = `
<section
  className="postsSidebar"
  id="sideBar"
>
  <h2
    className="postsSidebar__title"
  >
    Top Reddit Posts (/r/reactjs)
  </h2>
  <section
    className="postsList"
  >
    <ul
      className="pagination"
    >
      <li
        className="previous disabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="prev"
          role="button"
          tabIndex="0"
        >
          &lt;
        </a>
      </li>
      <li
        className="next"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="next"
          role="button"
          tabIndex="0"
        >
          &gt;
        </a>
      </li>
    </ul>
    <div
      className="postsList__container"
    >
      <div
        className="postsListItem"
      >
        <div
          className="postsListItem__row"
          onClick={[Function]}
        >
          <div
            className="postsListItem__author"
          >
            <img
              alt="Blue dot"
              className="postsListItem__author--indicator"
              src={Object {}}
            />
            <h2
              className="postsListItem__author--text"
            >
              0-07i
            </h2>
          </div>
          <p
            className="postsListItem__time"
          >
            NaN second ago
          </p>
        </div>
        <div
          className="postsListItem__row"
          onClick={[Function]}
        >
          <img
            alt=""
            className="postsListItem__thumb"
            src=""
          />
          <p
            className="postsListItem__description"
          >
            How do I find external services in the code?
          </p>
          <img
            alt="Right Arrow Icon"
            className="postsListItem__icon"
            src={Object {}}
          />
        </div>
        <div
          className="postsListItem__row"
        >
          <div
            className="postsListItem__dismiss"
            onClick={[Function]}
          >
            <img
              alt="Dismiss Icon"
              className="postsListItem__dismiss--icon"
              src={Object {}}
            />
            <p
              className="postsListItem__dismiss--text"
            >
              Dismiss Post
            </p>
          </div>
          <p
            className="postsListItem__comments"
            onClick={[Function]}
          >
            undefined comments
          </p>
        </div>
      </div>
      <div
        className="postsListItem"
      >
        <div
          className="postsListItem__row"
          onClick={[Function]}
        >
          <div
            className="postsListItem__author"
          >
            <img
              alt="Blue dot"
              className="postsListItem__author--indicator"
              src={Object {}}
            />
            <h2
              className="postsListItem__author--text"
            >
              junior19932
            </h2>
          </div>
          <p
            className="postsListItem__time"
          >
            NaN second ago
          </p>
        </div>
        <div
          className="postsListItem__row"
          onClick={[Function]}
        >
          <img
            alt=""
            className="postsListItem__thumb"
            src=""
          />
          <p
            className="postsListItem__description"
          >
            How do I find external services in the code?
          </p>
          <img
            alt="Right Arrow Icon"
            className="postsListItem__icon"
            src={Object {}}
          />
        </div>
        <div
          className="postsListItem__row"
        >
          <div
            className="postsListItem__dismiss"
            onClick={[Function]}
          >
            <img
              alt="Dismiss Icon"
              className="postsListItem__dismiss--icon"
              src={Object {}}
            />
            <p
              className="postsListItem__dismiss--text"
            >
              Dismiss Post
            </p>
          </div>
          <p
            className="postsListItem__comments"
            onClick={[Function]}
          >
            undefined comments
          </p>
        </div>
      </div>
      <div
        className="postsListItem"
      >
        <div
          className="postsListItem__row"
          onClick={[Function]}
        >
          <div
            className="postsListItem__author"
          >
            <img
              alt="Blue dot"
              className="postsListItem__author--indicator visited"
              src={Object {}}
            />
            <h2
              className="postsListItem__author--text visited"
            >
              aareedy
            </h2>
          </div>
          <p
            className="postsListItem__time"
          >
            NaN second ago
          </p>
        </div>
        <div
          className="postsListItem__row"
          onClick={[Function]}
        >
          <img
            alt=""
            className="postsListItem__thumb visited"
            src="https://a.thumbs.redditmedia.com/08MUc3eyJ4zs2f29UIAG53tGKMU6KwOFTb1An2aGUJ4.jpg"
          />
          <p
            className="postsListItem__description visited"
          >
            How do I find external services in the code?
          </p>
          <img
            alt="Right Arrow Icon"
            className="postsListItem__icon"
            src={Object {}}
          />
        </div>
        <div
          className="postsListItem__row"
        >
          <div
            className="postsListItem__dismiss"
            onClick={[Function]}
          >
            <img
              alt="Dismiss Icon"
              className="postsListItem__dismiss--icon"
              src={Object {}}
            />
            <p
              className="postsListItem__dismiss--text"
            >
              Dismiss Post
            </p>
          </div>
          <p
            className="postsListItem__comments"
            onClick={[Function]}
          >
            undefined comments
          </p>
        </div>
      </div>
    </div>
  </section>
  <h3
    className="postsSidebar__dismissAll"
    onClick={[Function]}
  >
    Dismiss All
  </h3>
  <img
    alt=""
    className="postsDetail__icon"
    id="btn"
    onClick={[Function]}
    src={Object {}}
  />
  <img
    alt=""
    className="postsDetail__icon--reload"
    onClick={[Function]}
    src={Object {}}
  />
</section>
`;
